modules/post/multi/manage/shell_to_meterpreter.rb

```diff
@@ -138,7 +140,18 @@ def run
       return nil
     end
 
-    payload_name = datastore['PAYLOAD_OVERRIDE'] if datastore['PAYLOAD_OVERRIDE']
+    if datastore['PAYLOAD_OVERRIDE']
+      payload_name = datastore['PAYLOAD_OVERRIDE']
+      payload_info = payload_name.split('/')
+      payload = framework.payloads.create(payload_name)
+
+      if payload_info.first == 'windows'
+        psh_arch = payload.arch
+      else
+        lplat = payload.platform.platforms
```

Comment:
Similar to the above - and I can't write a comment on the line here:

https://github.com/rapid7/metasploit-framework/blob/62a02e90b794d5632b407815c28446c781fa5933/modules/post/multi/manage/shell_to_meterpreter.rb#L172-L220

But I'm guessing from the original issue if you're using a PAYLOAD_OVERRIDE, I imagined the current `platform` is meant to be ignored entirely, and `lplat` is meant to be used in lieu of the session platform 🤔 

i.e. Does that logic also need to be changed too in some way? 👀